tests/conftest.py

```diff
@@ -28,29 +44,125 @@ def bucket_cleanup():
     return bucket_manager
 
 
-@pytest.fixture
-def unique_resource_name(request):
+@pytest.fixture(scope="session")
+def setup_nsfs_server_tls_cert():
     """
-    Generates a unique resource name with the given prefix
+    Configure the NSFS server TLS certification and download the certificate
+    in a local file.
+
+
+
+    Args:
+        config_root (str): The path to the configuration root directory.
+
+    Returns:
+        str: The path to the downloaded certificate file.
+
     """
+    conn = SSHConnectionManager().connection
+    config_root_path = get_config_root_full_path()
+    remote_credentials_dir = f"{config_root_path}/certificates"
+
+    # Create the TLS credentials and configure the NSFS service to use them
+    conn.exec_cmd(f"sudo mkdir -p {remote_credentials_dir}")
+    remote_tls_crt_path = create_tls_key_and_cert(remote_credentials_dir)
+    set_nsfs_service_certs_dir(remote_credentials_dir)
+    restart_nsfs_service()
 
-    def _get_unique_name(prefix="resource"):
-        unique_id = str(uuid.uuid4()).split("-")[0]
-        return f"{prefix}-{unique_id}"
+    # Download the certificate to a local file
+    with tempfile.NamedTemporaryFile(
+        prefix="tls_", suffix=".crt", delete=False
+    ) as local_tls_crt_file:
+        conn.download_file(
+            remotepath=remote_tls_crt_path,
+            localpath=local_tls_crt_file.name,
+        )
 
-    return _get_unique_name
+    S3Client.static_tls_crt_path = local_tls_crt_file.name
 
 
-@pytest.fixture(scope="function")
-def random_hex(request):
+@pytest.fixture
+def s3_client_factory(setup_nsfs_server_tls_cert, account_manager):
```

Comment:
add doc string

Reply:
done in https://github.com/red-hat-storage/noobaa-sa-ci/pull/11/commits/bdf79c5c8a8f0626f87f6e83694334e53086fdbf